File: src/main/java/shopping/product/infrastructure/persistence/jpa/ProductJpaRepository.java
package shopping.product.infrastructure.persistence.jpa;

import org.springframework.data.jpa.repository.JpaRepository;
import shopping.product.domain.ProductEntity;

import java.util.List;
import java.util.Optional;

public interface ProductJpaRepository extends JpaRepository<ProductEntity, Long> {
    Optional<ProductEntity> findByIdAndDeletedFalse(Long id);

    List<ProductEntity> findAllByDeletedFalse();
}

File: src/main/kotlin/shopping/product/infra/ProductJpaRepository.kt
package shopping.product.infra

import com.linecorp.kotlinjdsl.support.spring.data.jpa.repository.KotlinJdslJpqlExecutor
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import shopping.product.domain.Product

@Repository
interface ProductJpaRepository : JpaRepository<Product, Long>, KotlinJdslJpqlExecutor
